test/unit/test_program.py

```diff
@@ -139,3 +140,17 @@ def test_filter_quil_t():
     )
     full_program = non_quil_t_program + quil_t_program
     assert full_program.remove_quil_t_instructions() == non_quil_t_program
+
+
+def test_compatibility_layer():
+    """
+    Test that the compatibility layer that transforms pyQuil instructions to quil instructions works as intended.
+    """
+    # Note: `quil` re-orders some instructions in a program (e.g. by shuffling DECLAREs to the top). This isn't a
+    # problem for the current set of instructions we're testing, but it's something to keep in mind if we add more.
+    instructions = [ResetQubit(0), Reset()]
+    program = Program(instructions)
+    for (original, transformed) in zip(instructions, program):
```

Comment:
Didn't realize (or maybe forgot) that `Program` is an iterable over `Instruction`